lib/ocpp/common/websocket/websocket_uri.cpp

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Pionix GmbH and Contributors to EVerest
+
+#include <ocpp/common/types.hpp>
+#include <ocpp/common/websocket/websocket_uri.hpp>
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <websocketpp/uri.hpp>
+
+#include <stdexcept>
+#include <string>
+#include <string_view>
+
+namespace ocpp {
+
+auto path_split_last_segment(std::string path) {
+    struct result {
+        std::string path_without_last_segment;
+        std::string last_segment;
+    };
+
+    if (path == "/") {
+        return result{"/", ""};
+    }
+    auto pos_last_slash = path.rfind("/");
+
+    if (pos_last_slash == path.length() - 1) {
+        path.pop_back();
+        pos_last_slash -= 1;
+    }
+
+    auto path_without_last_segment = std::string(path.substr(0, pos_last_slash + 1));
+    std::string last_segment = std::string(path.substr(pos_last_slash + 1));
+
+    auto last_segment = std::string(path.substr(pos_last_slash + 1));
```

Comment:
you have `last_segment` twice here, which does not compile. Please fix ( I don't know how to request a change but this definitely is :))

Reply:
Yeah, sorry. I don't have any automated build running locally when committing stuff. Seems I've forgot it here.
But we're on adding build checks for all PR in CI :sweat_smile: